### PR TITLE
First ecrecover is tried and if that fails ERC-1271 is tried

### PR DESCRIPTION
**Pull request by gpersoon**
This is an alternative solution to https://github.com/ProjectOpenSea/seaport/pull/441

## Motivation
The current solution fails with Gnosis safe.
Verifying a the signer is a contract wallet is relative expensive, so this solution tries ecrecover first. 

## Solution
First ecrecover is tried and if that fails ERC-1271 is tried

### CODE REVIEW

File: contracts/lib/SignatureVerification.sol [hunk 1]
```diff
@@ -36,6 +33,31 @@ contract SignatureVerification is SignatureVerificationErrors, LowLevelHelpers {
         bytes32 digest,
         bytes memory signature
     ) internal view {
+        address recoveredSigner = _tryEcRecoverSignature(digest, signature);
```
Comment (hrkrshnn) [inline contracts/lib/SignatureVerification.sol:36]: I think it's better to add a check for `0` here. A bit worried if `signer` can be `0` for the call.

File: contracts/lib/SignatureVerification.sol [hunk 2]
```diff
@@ -83,31 +104,16 @@ contract SignatureVerification is SignatureVerificationErrors, LowLevelHelpers {
                     byte(v, ECDSA_twentySeventhAndTwentyEighthBytesSet)
                 )
             }
-
-            // Ensure v value is properly formatted.
+            // Ensure v value is properly formatted. Return invalid value if not
```
Comment (axic) [inline contracts/lib/SignatureVerification.sol:107]:
```suggestion
            // Early abort if v value is out of range to save the cost of ecrecover.
```

**Comment by axic** [comment]
> The current solution fails with Gnosis safe.

See our submission to C4 once released, but basically some signature modes supported by Safe do not work with the current code.

**Comment by gpersoon** [comment]
Closed as https://github.com/ProjectOpenSea/seaport/pull/441 is very similar now and has better error handling.

**Pull request closed without merging**